src/pip/_internal/req/req_uninstall.py

```diff
@@ -359,7 +359,7 @@ def from_dist(cls, dist):
             # develop egg
             with open(develop_egg_link, 'r') as fh:
                 link_pointer = os.path.normcase(fh.readline().strip())
-            assert (link_pointer == dist.location), (
+            assert (os.stat(link_pointer) == os.stat(dist.location)), (
```

Comment:
Should use `os.path.samefile`, as this is defined to do precisely this job. You may need to fall back to something else on Python 2.7, as samefile only added Windows support in Python 3.2, but I'm not sure if a stat comparison is sufficient on before Python 3.4 on Windows (the docs to samefile say "Changed in version 3.4: Windows now uses the same implementation as all other platforms" which implies that prior to that, Windows needed special treatment).

OTOH, this is "only" an assertion, not a normal error - so maybe it could be removed if it's asserting something that isn't always true? What do the other @pypa/pip-committers think?